src/background/api/index.js

```diff
@@ -2,18 +2,22 @@ import * as oasis from '@oasisprotocol/client';
 import { cointypes, TX_LIST_LENGTH } from "../../../config";
 import { amountDecimals, isNumber } from '../../utils/utils';
 import { commonFetch, getOasisClient } from "./request";
+import request from "./request"; // to make commonFetch jest.spyOn'able
+import rpc from './rpc'; // to make getRpcAccount jest.spyOn'able
 import * as oasisRT from '@oasisprotocol/client-rt';
+import getRpcAccount from './rpc'
 const MAX_LENGTH = 500
 /**
  * get account balance
  */
 export async function getBalance(address) {
   let url = "/chain/account/info/" + address
-  let account = await commonFetch(url).catch(() => { })
+  let account = await request.commonFetch(url).catch(() => { })
   if (account && account.code === 0) {
     return { ...account.data, address }
   } else {
-    return {}
+    let account = await rpc.getRpcAccount(address)
+    return { ...account.data, address }
```

Comment:
account.data is undefined. This can't work at all

Reply:
Here's how I debug this:
- inspect popup
- unlock
- F5 in devtools
- see request to https://api.oasisscan.com/mainnet/chain/account/info/oasis1qz78ap0456g2rk7j6rmtvasc9v2kjhz2s58qgj90
- rightclick, Block request URL
- F5

Reply:
I'm not sure what all the code using `getBalance` needs, but it currently returns e.g.
```ts
{
    "rank": 0,
    "address": "oasis1qz78ap0456g2rk7j6rmtvasc9v2kjhz2s58qgj90",
    "available": "119.625999998",
    "escrow": "0",
    "debonding": "0",
    "total": "119.625999998",
    "nonce": 11,
    "allowances": [
        {
            "address": "oasis1qrnu9yhwzap7rqh6tdcdcpz0zf86hwhycchkhvt8",
            "amount": "1"
        }
    ]
}
```

You can use `getRpcBalance` to get `nonce`, `allowances` with some parsing, and `available` balance string, with correction:
```diff
- balance = oasis.quantity.toBigInt(balance).toString()
+ balance = amountDecimals(oasis.quantity.toBigInt(balance)).toString()
```

You can use `oasisClient.stakingDelegationInfosFor` and `oasisClient.stakingDebondingDelegationInfosFor` to get `escrow` and `debonding` and sum `total`. Those are a bit heavy RPC calls tho and might fail; maybe another fallback to "?" if dependent code is okay with it.

Hopefully `rank` isn't needed.

Reply:
@lukaw3d I fixed that. Please take a look.

Reply:
I'm getting different values:
```js
// from API (testnet)
{
  "address": "oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km",
  "nonce": 33,
  "available": "7913.0999",
  "escrow": "100.973874127",
  "debonding": "0",
  "total": "8014.073774127",

  "allowances": [
    { "address": "oasis1qqdn25n5a2jtet2s5amc7gmchsqqgs4j0qcg5k0t", "amount": "3" },
    { "address": "oasis1qr629x0tg9gm5fyhedgs9lw5eh3d8ycdnsxf0run", "amount": "8" },
  ],
  "rank": 0,
}

// from fallback (testnet)
{
  "address": "oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km",
  "nonce": 0,
  "available": "7913.0999",
  "escrow": "100",
  "debonding": "0",
  "total": "8013.0999",

  "allowances": [],
  "rank": 0,
}
```

---

To fix:
- nonce: use from getRpcAccount
- escrow and debonding: Returns shares instead of amounts. Here's how web wallet parses it https://github.com/oasisprotocol/oasis-wallet-web/blob/a554cf974afc69ea27035d809b3649cde2d350b0/src/vendors/monitor.ts#L163-L187
- allowances and rank: I'd rather remove the fields and add a return type without them

Reply:
Now that I looked at what `getRpcAccount` returns it has everything already :/ 
<img width="610" alt="Screen Shot 2022-04-05 at 12 02 26" src="https://user-images.githubusercontent.com/48595253/161718963-f1c998a7-94c8-44ef-a5cf-8ce661adf42c.png">

Reply:
(for reference, from slack: that screenshot is API result, not RPC)

Reply:
Fixed!